File: tests/src/com.squareup.retrofit2/retrofit/2.5.0/src/test/java/com_squareup_retrofit2/retrofit/RetrofitAnonymous1Test.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_squareup_retrofit2.retrofit;

import okhttp3.ResponseBody;
import org.junit.jupiter.api.Test;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.http.GET;

import static org.assertj.core.api.Assertions.assertThat;

public class RetrofitAnonymous1Test {
    @Test
    void delegatesObjectMethodsOnServiceProxyToInvocationHandler() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("https://example.com/")
                .build();

        ProxyObjectMethodService service = retrofit.create(ProxyObjectMethodService.class);

        assertThat(service.toString()).isNotBlank();
    }

    public interface ProxyObjectMethodService {
        @GET("messages")
        Call<ResponseBody> messages();
    }
}

File: tests/src/com.squareup.retrofit2/retrofit/2.5.0/src/test/kotlin/com_squareup_retrofit2/retrofit/RetrofitAnonymous1Test.kt
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_squareup_retrofit2.retrofit

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.http.GET

public class RetrofitAnonymous1Test {
    public interface ObjectMethodService {
        @GET("users")
        public fun users(): Call<Void>
    }

    @Test
    public fun proxyObjectMethodDelegatesToInvocationHandlerObjectMethod(): Unit {
        val retrofit: Retrofit = Retrofit.Builder()
            .baseUrl("https://example.com/")
            .build()

        val service: ObjectMethodService = retrofit.create(ObjectMethodService::class.java)
        val proxyDescription: String = service.toString()

        assertThat(proxyDescription).isNotBlank()
    }
}
